common/src/main/java/io/netty/util/internal/logging/MessageFormatter.java

```diff
@@ -307,120 +284,139 @@ private static void safeObjectAppend(StringBuffer sbuf, Object o) {
-            if (i != len - 1) {
-                sbuf.append(", ");
+            if (i == iMax) {
+                return;
             }
+            sbuf.append(", ");
         }
-        sbuf.append(']');
     }
 
-    private static void shortArrayAppend(StringBuffer sbuf, short[] a) {
-        sbuf.append('[');
-        final int len = a.length;
-        for (int i = 0; i < len; i++) {
+    private static void shortArrayAppend(StringBuilder sbuf, short[] a) {
+        int iMax = a.length - 1;
+        if (iMax == -1) {
+            return;
+        }
+        for (int i = 0; ; i++) {
             sbuf.append(a[i]);
-            if (i != len - 1) {
-                sbuf.append(", ");
+            if (i == iMax) {
+                return;
             }
+            sbuf.append(", ");
         }
-        sbuf.append(']');
     }
 
-    private static void intArrayAppend(StringBuffer sbuf, int[] a) {
-        sbuf.append('[');
-        final int len = a.length;
-        for (int i = 0; i < len; i++) {
+    private static void intArrayAppend(StringBuilder sbuf, int[] a) {
+        int iMax = a.length - 1;
+        if (iMax == -1) {
+            return;
+        }
+        for (int i = 0; ; i++) {
             sbuf.append(a[i]);
-            if (i != len - 1) {
-                sbuf.append(", ");
+            if (i == iMax) {
+                return;
             }
+            sbuf.append(", ");
         }
-        sbuf.append(']');
     }
 
-    private static void longArrayAppend(StringBuffer sbuf, long[] a) {
-        sbuf.append('[');
-        final int len = a.length;
-        for (int i = 0; i < len; i++) {
+    private static void longArrayAppend(StringBuilder sbuf, long[] a) {
+        int iMax = a.length - 1;
+        if (iMax == -1) {
+            return;
+        }
+        for (int i = 0; ; i++) {
             sbuf.append(a[i]);
-            if (i != len - 1) {
-                sbuf.append(", ");
+            if (i == iMax) {
+                return;
             }
+            sbuf.append(", ");
         }
-        sbuf.append(']');
     }
 
-    private static void floatArrayAppend(StringBuffer sbuf, float[] a) {
-        sbuf.append('[');
-        final int len = a.length;
-        for (int i = 0; i < len; i++) {
+    private static void floatArrayAppend(StringBuilder sbuf, float[] a) {
+        int iMax = a.length - 1;
+        if (iMax == -1) {
+            return;
+        }
+        for (int i = 0; ; i++) {
             sbuf.append(a[i]);
-            if (i != len - 1) {
-                sbuf.append(", ");
+            if (i == iMax) {
+                return;
             }
+            sbuf.append(", ");
         }
-        sbuf.append(']');
     }
 
-    private static void doubleArrayAppend(StringBuffer sbuf, double[] a) {
-        sbuf.append('[');
-        final int len = a.length;
-        for (int i = 0; i < len; i++) {
+    private static void doubleArrayAppend(StringBuilder sbuf, double[] a) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)